stdlib/asm/word.masm

```diff
@@ -0,0 +1,228 @@
+
+        movdn.3
+        # => [2_x, 1_x, is_lhs_greater, is_felt_eq, continue, <remaining_felts>]
+
+        # check 1_x > 2_x
+        gt
+        # => [is_felt_gt, is_lhs_greater, is_felt_eq, continue, <remaining_felts>]
+
+        dup.3 and
+        # => [is_felt_gt_if_continue, is_lhs_greater, is_felt_eq, continue, <remaining_felts>]
+
+        or movdn.2
+        # => [is_felt_eq, continue, is_lhs_greater, <remaining_felts>]
+
+        # keeps continue at 1 if the felts are equal
+        # sets continue to 0 if the felts are not equal
+        and
+        # => [continue, is_lhs_greater, <remaining_felts>]
+
+        swap
+        # => [is_lhs_greater, continue, <remaining_felts>]
+    end
+    # => [is_lhs_greater, continue]
+
+    swap drop
+    # => [is_lhs_greater]
+end
+
+#! Returns true if LHS is greater than or equal to RHS.
+#!
+#! Inputs:  [LHS, RHS]
+#! Output: [is_lhs_greater_or_equal]
+#!
+#! Cycles: 118
+export.gte
+    exec.lt
+    not
+end
+
+#! Returns true if LHS is strictly less than RHS, false otherwise.
+#!
+#! The implementation avoids branching for performance reasons.
+#!
+#! From an implementation standpoint this is exactly the same as `word::gt` except it uses
+#! `lt` rather than `gt`. See its docs for details.
+#!
+#! Inputs:  [LHS, RHS]
+#! Output: [is_lhs_lesser]
+#!
+#! Cycles: 117
+export.lt
+    exec.arrange_words_adjacent
+    # => [2_3, 1_3, 2_2, 1_2, 2_1, 1_1, 2_0, 1_0]
+
+    push.1.0
+    # => [is_lhs_less, continue, 2_3, 1_3, 2_2, 1_2, 2_1, 1_1, 2_0, 1_0]
+
+    repeat.4
+        movup.3 movup.3
+        # => [2_x, 1_x, is_lhs_less, continue, <remaining_felts>]
+
+        # check 1_x == 2_x; if so, we continue
+        dup dup.2 eq
+        # => [is_felt_eq, 2_x, 1_x, is_lhs_less, continue, <remaining_felts>]
+
+        movdn.3
+        # => [2_x, 1_x, is_lhs_less, is_felt_eq, continue, <remaining_felts>]
+
+        # check 1_x < 2_x
+        lt
+        # => [is_felt_lt, is_lhs_less, is_felt_eq, continue, <remaining_felts>]
+
+        dup.3 and
+        # => [is_felt_lt_if_continue, is_lhs_less, is_felt_eq, continue, <remaining_felts>]
+
+        or movdn.2
+        # => [is_felt_eq, continue, is_lhs_less, <remaining_felts>]
+
+        # keeps continue at 1 if the felts are equal
+        # sets continue to 0 if the felts are not equal
+        and
+        # => [continue, is_lhs_less, <remaining_felts>]
+
+        swap
+        # => [is_lhs_less, continue, <remaining_felts>]
+    end
+    # => [is_lhs_less, continue]
+
+    swap drop
+    # => [is_lhs_less]
+end
+
+#! Returns true if LHS is less than or equal to RHS.
+#!
+#! Inputs:  [LHS, RHS]
+#! Output: [is_lhs_less_or_equal]
+#!
+#! Cycles: 122
+export.lte
+    exec.gt
+    not
+end
+
+#! Returns true if LHS is exactly equal to RHS, false otherwise.
+#!
+#! The implementation does not branch, and always performs the same number of comparisons.
+#!
+#! This is currently equivalent to the eqw instruction.
```

Comment:
Actually, `eqw` instruction works differently - it keeps both words on the stack. Also, the implementation is:
```
dup.7, dup.4, eq,
dup.7, dup.4 eq, and,
dup.6, dup.3, eq, and,
dup.5, dup.2, eq, and,
```
which is 15 cycles.

I think for the `eq` procedure we should use the following implementation:
```
movup.4 eq swap
movup.4 eq and swap
movup.3 eq and
movdn.2 eq and
```
Which should be 13 cycles.

And I would also add `test_eq` procedure which don't consume the inputs and used the implementation equivalent to `eqw`.